src/datasources/asset/components/editors/asset-summary/AssetSummaryDataSource.ts

```diff
@@ -17,10 +19,29 @@ export class AssetSummaryDataSource extends AssetDataSourceBase {
     defaultQuery = {
     };
 
-    runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-        throw new Error("Method not implemented.");
+    async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+        return this.processMetadataQuery(query as AssetQuery);
+    }
+
+    shouldRunQuery(_: AssetQuery): boolean {
+        return true;
     }
-    shouldRunQuery(query: AssetQuery): boolean {
-        throw new Error("Method not implemented.");
+
+    async processMetadataQuery(query: AssetQuery) {
+        const result: DataFrameDTO = { refId: query.refId, fields: [] };
```

Comment:
nit: I don't see the point of defining this variable and the setting the fields in two steps. Just return the whole thing at the end:

```ts
return {
  refId: query.refId,
  fields: [
    { name: assetSummaryFields.TOTAL, values: [assets.total] },
    ...
  ]
}
```